python/src/acp_sdk/server/server.py

```diff
@@ -241,3 +251,46 @@ def should_exit(self) -> bool:
     @should_exit.setter
     def should_exit(self, value: bool) -> None:
         self._server.should_exit = value
+
+    async def _register_agent(self) -> None:
```

Comment:
Please hide this behind `self_registration` flag that is on by default. (similar to logging and telemetry)